### PR TITLE
add struct util functions and default struct tags

### PR DESCRIPTION
**Pull request by davidbloss**
## Changelog

Add `default:""` struct tags to:
- `MemberInput`, `PropertyDefinitionInput`, `ToolCreateInput`, `ToolUpdateInput`, `UserInput`, `UserIdentifierInput`

Add `utils.go` with these functions:
- `IsResourceValidate[T any](opslevelResource T)` - validates structs based on `validate:""` struct tags
- `SetDefaultsFor[T any](opslevelResource *T)` - sets defaults defined by `default:""` struct tags
- `NewExampleOf[T any]()` - generic constructor that sets defaults defined by `default:""` struct tags
- `JsonOf[T any](opslevelResource T)` - converts struct to JSON formatted string
- `YamlOf[T any](opslevelResource T)` - converts struct to Yaml formatted string

`IsResourceValidate` is not yet used

- [X] List your changes here
- [X] Make a `changie` entry

## Tophatting

`task test` - tests pass

### CODE REVIEW

File: tools_test.go [hunk 1]
```diff
@@ -9,19 +9,15 @@ import (
 
 func TestCreateTool(t *testing.T) {
 	// Arrange
+	toolCreateInput := autopilot.Register[ol.ToolCreateInput]("tool_create_input", ol.NewExampleOf[ol.ToolCreateInput]())
```
Comment (rocktavious) [inline tools_test.go:12]: OMG 🚀

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	"github.com/creasty/defaults"
+	"github.com/go-playground/validator/v10"
+	"gopkg.in/yaml.v3"
+)
+
+var structValidator = validator.New(validator.WithRequiredStructEnabled())
+
+// Validates resource's `validate:""` struct tags
+func IsResourceValidate[T any](opslevelResource T) error {
```
Comment (rocktavious) [inline utils.go:14]: These function names are 😍  - Great Job!

**Review by rocktavious — APPROVED** [review]
Amazing!

File: utils.go (hunk 2 above)
Comment (davidbloss) [reply utils.go:14]: 🤣

**Pull request merged**